libsolidity/formal/CHC.cpp

```diff
@@ -891,41 +891,35 @@ void CHC::defineInterfacesAndSummaries(SourceUnit const& _source)
 			auto const& iface = *m_nondetInterfaces.at(contract);
 			addRule(smt::nondetInterface(iface, *contract, m_context, 0, 0), "base_nondet");
 
-			for (auto const* base: contract->annotation().linearizedBaseContracts)
-				for (auto const* function: base->definedFunctions())
+			for (auto const* function: contractFunctions(*contract))
+			{
+				for (auto var: function->parameters())
+					createVariable(*var);
+				for (auto var: function->returnParameters())
+					createVariable(*var);
+				for (auto const* var: localVariablesIncludingModifiers(*function, contract))
+					createVariable(*var);
+
+				m_summaries[contract].emplace(function, createSummaryBlock(*function, *contract));
+
+				if (!function->isConstructor() && function->isPublic())
```

Comment:
We don't have the ability to check `base->isLibrary` and `base->isInterface`, but I have not seen any problems in the tests.

Reply:
Yea, I think those conditions were redundant since it's not possible to inherit from a library or an interface. The latter might change in the future though.